Autopilot/PathManager/Inc/waypointManager.hpp

```diff
@@ -0,0 +1,242 @@
+/**
+ * Waypoint Manager Methods and Helpers
+ * Author: Dhruv Rawat
+ * Created: November 2020
+ * Last Updated: November 2020 (Dhruv)
+ */
+
+#ifndef WAYPOINT_MANAGER_H
+#define WAYPOINT_MANAGER_H
+
+#include <iostream>
+#include <stdlib.h>
+#include <math.h>
+
+#define ORBIT_FOLLOWING 0
+#define LINE_FOLLOWING 1
+
+//Constants
+#define PI 3.14159265
+#define EARTH_RADIUS 6378.137
+#define PATH_BUFFER_SIZE 100
+#define MAX_PATH_APPROACH_ANGLE PI/2
+
+//Basic Mathematical Conversions
+#define deg2rad(angle_in_degrees) ((angle_in_degrees) * PI/180.0)
+#define rad2deg(angle_in_radians) ((angle_in_radians) * 180.0/PI)
+
+//Waypoint Types
+#define DEFAULT_WAYPOINT 0 // Used to navigate flight path
+#define WAYPOINT_USED 1 // Used by the waypint manager to signal that a waypoint has been used
+#define HOLD_WAYPOINT 2 // Circle
+
+// Latitude and longitude of starting area... will need to change accordingly
+#define RELATIVE_LATITUDE 43.473004
+#define RELATIVE_LONGITUDE -80.539678
```

Comment:
Could these be defined in a constructor of sort?

Reply:
defined is the wrong word, set is better LOL

Reply:
Yes, these could be set in the constructor using gps data.

Regarding your original comment above:
1. If you want the plane to start circling/heading home, call these functions. What they will do is they will trigger a Boolean variable.
2. continue calling get_next_direction() as normal. Because of the Boolean variables being triggered, there will be an if statement that will be triggered that will make the plane not follow the waypointBuffer array and instead circle, or head towards the waypoint representing home base. 
3. When you want the plane to start following the waypointBuffer array, you can call the start_circling/head_home methods. 

In short, the two methods are like setter methods that set a Boolean variable that makes the plane execute the get_next_directions() method differently.